process_report/tests/unit_tests.py

```diff
@@ -175,3 +176,101 @@ def test_export_pi(self):
+
+@skipIf(process_report.apply_credits_0002 not in process_report.applied_credits, "Skipping test for credit 0002 because credit not enabled")
+class TestCredit0002(TestCase):
+    def setUp(self):
+
+        data = {
+            'Manager (PI)': ['PI1', 'PI1', 'PI2', 'PI3', 'PI4', 'PI4'],
+            'Project - Allocation': ['ProjectA', 'ProjectB', 'ProjectC', 'ProjectD', 'ProjectE', 'ProjectF'],
+            'Cost': [10, 100, 10000, 5000, 800, 1000]
+        }
+        self.dataframe = pandas.DataFrame(data)
+        old_pi = ['PI2', 'PI3']
+        old_pi_file = tempfile.NamedTemporaryFile(delete=False, mode='w', suffix='.csv')
+        for pi in old_pi: old_pi_file.write(pi + "\n")
+        self.old_pi_file = old_pi_file.name
+
+        os.environ["C0002_OLD_PI"] = self.old_pi_file
+
+    def tearDown(self):
+        os.remove(self.old_pi_file)
+
+    def test_apply_credit_0002(self):
+        dataframe = process_report.add_credits(self.dataframe)
+
+        self.assertTrue('Credit' in dataframe)
+        self.assertTrue('Credit Code' in dataframe)
+        self.assertTrue('Balance' in dataframe)
+
+        credited_projects = dataframe[dataframe['Credit Code'] == '0002']
+
+        self.assertEqual(4, len(credited_projects.index))
+        self.assertTrue('PI2' not in credited_projects['Manager (PI)'].unique())
+        self.assertTrue('PI3' not in credited_projects['Manager (PI)'].unique())
+
+        self.assertEqual(10, credited_projects[credited_projects['Project - Allocation'] == 'ProjectA']['Credit'].iloc[0])
+        self.assertEqual(100, credited_projects[credited_projects['Project - Allocation'] == 'ProjectB']['Credit'].iloc[0])
+        self.assertEqual(800, credited_projects[credited_projects['Project - Allocation'] == 'ProjectE']['Credit'].iloc[0])
+        self.assertEqual(200, credited_projects[credited_projects['Project - Allocation'] == 'ProjectF']['Credit'].iloc[0])
+
+        self.assertEqual(0, credited_projects[credited_projects['Project - Allocation'] == 'ProjectA']['Balance'].iloc[0])
+        self.assertEqual(0, credited_projects[credited_projects['Project - Allocation'] == 'ProjectB']['Balance'].iloc[0])
+        self.assertEqual(0, credited_projects[credited_projects['Project - Allocation'] == 'ProjectE']['Balance'].iloc[0])
+        self.assertEqual(800, credited_projects[credited_projects['Project - Allocation'] == 'ProjectF']['Balance'].iloc[0])
+
+
+class TestExportLenovo(TestCase):
+    def setUp(self):
+
+        data = {
+            'Invoice Month': ['2023-01','2023-01','2023-01','2023-01','2023-01', '2023-01'],
+            'Project - Allocation': ['ProjectA', 'ProjectB', 'ProjectC', 'ProjectD', 'ProjectE', 'ProjectF'],
+            'Institution': ['A', 'B', 'C', 'D', 'E', 'F'],
+            'SU Hours (GBhr or SUhr)': [1, 10, 100, 4, 432, 10],
+            'SU Type': ['OpenShift GPUA100SXM4', 'CPU', 'OpenShift GPUA100SXM4', 'OpenStack GPUA100SXM4', 'CPU', 'GPU']
```

Comment:
instead of GPU can we put GPUA100, and instead of a second CPU can we put GPUV100

Reply:
@joachimweyl  For the purpose of this test, I don't think replacing the string `GPU` with `GPUA100` would make a difference, since to know if a project used a Lenovo SU type, the code tries to match the SU Type exactly to either `OpenShift GPUA100SXM4` or `OpenStack GPUA100SXM4`

Reply:
It is not supposed to, but the point of a test is to test all possibilities right?
To be clear we should put `OpenShift GPUA100` and `OpenStack GPUV100`. and `CPU` should actually be `OpenShift CPU` or `OpenStack CPU`.  We also might want to test that it properly combines `OpenStack GPUA100SXM4` so maybe replace one of the other ones with that instead.

Reply:
@joachimweyl What do you mean by properly combines `OpenStack GPUA100SXM4`?

Reply:
"calculates" not combines, sorry.

Reply:
I've updated the list of SU Types being tested.